source/type/data.cpp

```diff
@@ -430,13 +435,25 @@ static void build_datasets(navitia::type::VehicleJourney* vj) {
     }
 }
 
+static void build_companies(navitia::type::VehicleJourney* vj) {
+    if (vj->company) {
+        if (boost::range::find(vj->route->line->company_list, vj->company) == vj->route->line->company_list.end()) {
```

Comment:
minor - It would really ease readability to extract out things in separate variables, like:
```c++
Company* company = vj->company;
 if (company) {
        Line* line = vj->route->line;
        auto & company_list = line->company_list;
        if (boost::range::find(company_list, company) == company_list.end()) {
            company_list.push_back(company);
        }
        if (boost::range::find(company->line_list, line) == company->line_list.end()) {
            company->line_list.push_back(line);
        }
```